listenbrainz/replay_user.py

```diff
@@ -0,0 +1,93 @@
+""" This script should be used to replay user listens to fix bad data.
+"""
+import json
+import uuid
+
+from datetime import datetime
+from flask import current_app
+from listenbrainz.utils import get_measurement_name, quote, convert_to_unix_timestamp
+from listenbrainz.webserver import create_app
+from listenbrainz.webserver.influx_connection import init_influx_connection
+from listenbrainz.listenstore.influx_listenstore import DUMP_CHUNK_SIZE
+
+
+class UserReplayer:
+    def __init__(self, user_name):
+        self.user_name = user_name
+        self.max_time = datetime.now()
+        self.app = create_app()
+
+
+    def filter_function(self, row):
+        return row
+
+
+    def convert_to_influx_insert_format(self, row, measurement):
+        data = {
+            'measurement': measurement,
+            'time': convert_to_unix_timestamp(row['time']),
+        }
+
+        if 'dedup_tag' in row:
+            data['tags'] = {'dedup_tag': row['dedup_tag']}
+
+        data['fields'] = {}
+        for field in row:
+            if field != 'time' and field != 'dedup_tag':
+                data['fields'][field] = row[field]
+
+        return data
+
+
+    def copy_measurement(self, src, dest, apply_filter=False):
```

Comment:
Why would we not want to apply a filter?

Reply:
We do A -> B with filter, delete A, then do B -> A without filter, and delete B. the B -> A copy would not need a filter.

Reply:
Got it -- thanks!